torch/_inductor/codegen/wrapper.py

```diff
@@ -798,6 +798,19 @@ def write_header(self) -> None:
             )
         except (AttributeError, ImportError):
             pass
+        if torch.backends.mps.is_built():
+            self.header.splice(
+                """
+                def _compile_mps_shader(source):
+                    try:
+                        return torch.mps._compile_shader(source)
+                    except SyntaxError as err:
+                        raise SyntaxError(f"failed to compile {source} with {err.msg}")
+
+
+                """,
+                strip=True,
+            )
```

Comment:
Rather than sticking this in the header, can we put it in `torch._inductor.runtime.*` and import it?

Reply:
Have a look at the updated code